File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        vector<vector<int>> ans;
        vector<int> temp;
        recurse(nums, temp, ans, 0);
        return ans;
    }
    
    void recurse(vector<int>&nums, vector<int>&temp,  vector<vector<int>>& ans, int index ) {
        ans.push_back(temp);
        for(int i=index;i<nums.size();i++) {
            if(i!=index && nums[i]==nums[i-1]) continue;
            temp.push_back(nums[i]);
            recurse(nums,temp,ans,i+1);
            temp.pop_back();
        }
    }
};

File: 0090-subsets-ii/0090-subsets-ii.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsetsWithDup = function(nums) {
    let ans = [], temp = [];
    nums.sort();
    recurse(0);
    return ans;
    
    function recurse(index) {
        ans.push(temp.slice());
        for(let i = index; i < nums.length; i++) {
            if(i!=index && nums[i]===nums[i-1]) continue;
            temp.push(nums[i]);
            recurse(i+1);
            temp.pop();
        }
    }
    
};